tests-gtest/a_imugamma5.cc

```diff
@@ -0,0 +1,124 @@
+  static int constexpr soalen = 4;
+  static bool constexpr compress12 = true;
+
+  int latt_size_[4] = {16, 16, 16, 16};
+
+  Geometry<FT, veclen, soalen, compress12> geom_;
+  HybridSpinor<FT, veclen, soalen, compress12, QDP::LatticeFermionD> source_;
+  HybridSpinor<FT, veclen, soalen, compress12, QDP::LatticeFermionD> source_dn_;
+  HybridSpinor<FT, veclen, soalen, compress12, QDP::LatticeFermionD> step1_;
+  HybridSpinor<FT, veclen, soalen, compress12, QDP::LatticeFermionD> step1_dn_;
+  HybridSpinor<FT, veclen, soalen, compress12, QDP::LatticeFermionD> step2_;
+  HybridSpinor<FT, veclen, soalen, compress12, QDP::LatticeFermionD> step2_dn_;
+
+  typename Geometry<FT, veclen, soalen, compress12>::FourSpinorBlock *two_flav_source_[2];
+  typename Geometry<FT, veclen, soalen, compress12>::FourSpinorBlock *two_flav_step1_[2];
+  typename Geometry<FT, veclen, soalen, compress12>::FourSpinorBlock *two_flav_step2_[2];
+};
+
+/**
+  Verifies that multiplying with the inverse returns the original spinor.
+
+  We want to have
+  \f[
+    \psi = (\alpha + \mathrm i \mu \gamma_5)^{-1}
+    (\alpha + \mathrm i \mu \gamma_5) \psi \,.
+  \f]
+  The inverse is given by \f$
+    (\alpha + \mathrm i \mu \gamma_5)^{-1}
+    = \frac{\alpha}{\alpha^2 + \mu^2}
+    + \frac{- \mathrm i \mu \gamma_5}{\alpha^2 + \mu^2} \,.
+  \f$
+  */
+TEST_F(BlasTest, apimugamma5Inverse)
+{
+  double const alpha = 3.4;
+  double const mu = 1.2;
+  double const denominator = alpha * alpha + mu * mu;
+  double const apimu[] = {alpha, mu};
+  double const apimu_inv[] = {alpha / denominator, -mu / denominator};
+
+  twisted_mass(apimu, source_[0], step1_[0], geom_, 1);
+  twisted_mass(apimu_inv, step1_[0], step2_[0], geom_, 1);
+
+  step2_.unpack();
+  expect_near(source_, step2_, 1e-10, geom_, 0, "α+iμγ_5 normal and inverse");
+}
+
+TEST_F(BlasTest, twoFlavTwistedMassSignChange) {
+    double const alpha = 2.3;
+    double const mu = 4.2;
+    double const epsilon = 0.37;
+
+    double const apimu[2] = {alpha, mu};
+    double const apimu_minus[2] = {-alpha, -mu};
+
+    two_flav_twisted_mass(
+        apimu, epsilon, two_flav_source_, two_flav_step1_, geom_, 1);
+    two_flav_twisted_mass(
+        apimu_minus, -epsilon, two_flav_source_, two_flav_step2_, geom_, 1);
+
+    axpy<FT, veclen, soalen, compress12, 2>(
+        1.0, two_flav_step1_, two_flav_step2_, geom_, 1);
+
+    step2_.unpack();
+    step2_dn_.unpack();
+    expect_near(step1_, step2_, 1e-10, geom_, 0, "α+iμγ_5 plus and minus (up)");
+    expect_near(step2_dn_, step2_dn_, 1e-10, geom_, 0, "α+iμγ_5 plus and minus (down)");
+}
+
+TEST_F(BlasTest, twoFlavTwistedMassInverse) {
+    double const alpha = 3.0;
+    double const mu = 5.0;
+    double const epsilon = 7.0;
+    double const denominator = alpha * alpha + mu * mu - epsilon * epsilon;
+
+    double const apimu[2] = {alpha, mu};
+    double const apimu_inv[2] = {alpha / denominator, -mu / denominator};
+    double const epsilon_inv = -epsilon / denominator;
```

Comment:
But the tau1_scaleadd should definitely do plus! In the two-flavour EO operator, the minus sign for the off-diagonal bit is added by AChiMinusBDPsi.